ml/src/test/scala/frameless/ml/classification/TypedRandomForestClassifierTests.scala

```diff
@@ -0,0 +1,76 @@
+package frameless
+package ml
+package classification
+
+import shapeless.test.illTyped
+import org.apache.spark.ml.linalg._
+import TypedRandomForestClassifier.FeatureSubsetStrategy
+import org.scalacheck.{Arbitrary, Gen}
+import org.scalacheck.Prop._
+import org.scalatest.MustMatchers
+
+class TypedRandomForestClassifierTests extends FramelessMlSuite with MustMatchers {
+  implicit val arbDouble: Arbitrary[Double] =
+    Arbitrary(Gen.choose(1, 99).map(_.toDouble)) // num classes must be between 0 and 100
+  implicit val arbVectorNonEmpty: Arbitrary[Vector] =
+    Arbitrary(Generators.arbVector.arbitrary suchThat (_.size > 0)) // vector must not be empty for RandomForestClassifier
+
+  test("fit() returns a correct TypedTransformer") {
+    val prop = forAll { x2: X2[Double, Vector] =>
+      val rf = TypedRandomForestClassifier.create[X2[Double, Vector]]()
```

Comment:
Aesthetics suggestion: perhaps `TypedRandomForestClassifier[X2[Double, Vector]]` would be more ergonomic? E.g. `apply` method with no explicit parameter list

Reply:
Agreed.